mllib/src/main/scala/org/apache/spark/mllib/rdd/MLPairRDDFunctions.scala

```diff
@@ -47,7 +47,18 @@ class MLPairRDDFunctions[K: ClassTag, V: ClassTag](self: RDD[(K, V)]) extends Se
       combOp = (queue1, queue2) => {
         queue1 ++= queue2
       }
-    ).mapValues(_.toArray.sorted(ord.reverse))  // This is a min-heap, so we reverse the order.
+    ).mapValues(_.toArray.sorted(ord.reverse)) // This is a min-heap, so we reverse the order.
+  }
+
+  def topByKey(num: Int, bucketsCount: Int)(implicit ord: Ordering[V]): RDD[(K, Array[V])] = {
```

Comment:
You can just keep one method with a new optional Int argument. Its default can be `Partitioner.defaultPartitioner(self)` This new method isn't used though?